app/Http/Requests/StoringDoctorWorkingDayRequest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Enums\WeekDays;
+use BenSampo\Enum\Rules\EnumValue;
+use Illuminate\Foundation\Http\FormRequest;
+
+class StoringDoctorWorkingDayRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'day' => ['required', new EnumValue(WeekDays::class, false)],
+            'from' => 'nullable|required_with:to|date_format:g:a',
+            'to' => 'nullable|required_with:from|date_format:g:a|after:from',
```

Comment:
The doctor will send his working days on one request. Allow him to send an array of objects contains (day, from, to, all_day), validate them and store them in the database. It's not efficient to make user add his working shifts a shift by shift.

Reply:
done